clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java

```diff
@@ -289,4 +310,20 @@ Optional<Selector> getRelevantSelector(String methodIdentifier) {
     memo.putIfAbsent(methodIdentifier, mostApplicableSelector);
     return mostApplicableSelector;
   }
+
+  /**
+   * Get the justification token out of the headers, and create a struct of the correct format for
+   * use in audit logging metadata. We expect the claims from the JWT to be put in JSON format and
+   * put into the struct under the justification_token key.
+   */
+  Struct getStructForJustification(Metadata headers) throws JwkException {
+    Metadata.Key<String> metadataKey =
+        Metadata.Key.of("justification_token", Metadata.ASCII_STRING_MARSHALLER);
+    String jwtString = headers.get(metadataKey);
+    DecodedJWT jwt = jvs.validateJWT(jwtString);
+    String jsonString = StringUtils.newStringUtf8(Base64.decodeBase64(jwt.getPayload()));
```

Comment:
I don't know a lot about Java, but does this handle all the possible b64 encoding schemes automatically?

Reply:
Yes, it should decode to UTF8 all base64 encoded strings. https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base64.html

Reply:
I see:

> The URL-safe parameter is only applied to encode operations. Decoding seamlessly handles both modes.

But I don't see anything about decoded padded vs unpadded. Maybe let's add a test with a few different base64 formats to be sure?

These should all decode to `foor4?-`:

```text
Zm9vcjQ/LQo=
Zm9vcjQ_LQo=
Zm9vcjQ/LQo
Zm9vcjQ_LQo
```

Reply:
I put together a quick test.

```
 public static void main(String[] args) {
    List<String> strings = new ArrayList<>();
    String compareTo = StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ/LQo="));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ_LQo=")));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ/LQo")));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ_LQo")));

    for (String string : strings) {
      if (!compareTo.equals(string)) {
        throw new IllegalStateException("NOT EQUAL!");
      }
    }
  }
```

I can confirm the above code doesn't throw an exception, and if i print the strings, they all end up as the string `foor4?-`